### PR TITLE
Update http_microhttpd.c

### PR DESCRIPTION
**Pull request by jiqianghu**
support port 80

### CODE REVIEW

**Comment by mwarning** [comment]
I do not have the insight into this mechanism to know if this works correctly.
@lynxis any idea?

**Comment by lynxis** [comment]
@jiqianghu please improve your commit message. The summary (first line) should contain more information than update this or that file. It should also describe why your commit was necessary.
We usually prefix the topic by the filename or the component you changed. by "microhttpd: bla".
A good description how commit messages should look like is https://who-t.blogspot.de/2009/12/on-commit-messages.html

**Comment by lynxis** [comment]
@mwarning I guess we don't support nodogsplash to listen on port 80.

**Comment by lynxis** [comment]
related to #159

File: src/http_microhttpd.c [hunk 1]
```diff
@@ -151,8 +151,11 @@ static int is_splashpage(const char *host, const char *url)
 	} else {
 		/* hostname give - check if it's our hostname */
 
-		if (strcmp(host, our_host)) {
-			/* hostname isn't ours */
+		/* check host with port */
+		if (strcmp(host, our_host) && (80 != config->gw_port)) {
+			return 0;
+		}
+        	else if (strcmp(host, config->gw_address) && (80 == config->gw_port)) {
```
Comment (lynxis) [inline src/http_microhttpd.c]: Port 80 is more special, because both hostnames are valid:
* 192.168.1.1:80
* 192.168.1.1

Comment (mwarning) [inline src/http_microhttpd.c]: @lynxis it is not forbidden by the docs. But I think it is a good idea to not allow it until it is understood to cause no problems.

Comment (lynxis) [inline src/http_microhttpd.c]: @mwarning that's the reason, why I mentioned it. we have to support both cases in this function.

**Pull request closed without merging**